pandas/core/groupby/generic.py

```diff
@@ -270,9 +270,9 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
             func = maybe_mangle_lambdas(func)
             ret = self._aggregate_multiple_funcs(func)
             if relabeling:
-                # error: Incompatible types in assignment (expression has type
-                # "Optional[List[str]]", variable has type "Index")
-                ret.columns = columns  # type: ignore[assignment]
+                # columns is not narrowed by mypy from relabeling flag
+                assert columns is not None  # for mypy
```

Comment:
not yet ran the tests locally to check we don't get failures but it appears `relabeling` and `columns` are not independent variables, which mypy cannot understand and we should try to avoid. Not planning to refactor at this time.